src/nupic/types/Sdr.cpp

```diff
@@ -347,67 +347,13 @@ namespace sdr {
 
 
     void SparseDistributedRepresentation::save(std::ostream &outStream) const {
-
-        auto writeVector = [&outStream] (const vector<UInt> &vec) {
-            outStream << vec.size() << " ";
-            for( auto elem : vec ) {
-                outStream << elem << " ";
-            }
-            outStream << endl;
-        };
-
-        // Write a starting marker and version.
-        outStream << "SDR " << SERIALIZE_VERSION << " " << endl;
-
-        // Store the dimensions.
-        writeVector( dimensions );
-
-        // Store the data in the flat-sparse format.
-        writeVector( getSparse() );
-
-        outStream << "~SDR" << endl;
+        cereal::BinaryOutputArchive archive( outStream );
+        archive( *this );
```

Comment:
the logic is in save_ar, so this wrapper should be already implemented in Serializable. 
Or Serializable dictates the cereal method.

Reply:
Here it should be cereal::JSONOutputArchive for text.  Serializable class determines binary output.